cloud/services/resourceskus/cache_test.go

```diff
@@ -0,0 +1,535 @@
+			}
+		})
+	}
+}
+
+func TestCacheGetZones(t *testing.T) {
+	cases := map[string]struct {
+		have []compute.ResourceSku
+		want []string
+	}{
+		"should find 1 result": {
+			have: []compute.ResourceSku{
+				{
+					Name: to.StringPtr("foo"),
+					Kind: to.StringPtr(string(VirtualMachines)),
+					Locations: &[]string{
+						"baz",
+					},
+					LocationInfo: &[]compute.ResourceSkuLocationInfo{
+						{
+							Location: to.StringPtr("baz"),
+							Zones:    &[]string{"1"},
+						},
+					},
+				},
+			},
+			want: []string{"1"},
+		},
+		"should find 2 results": {
+			have: []compute.ResourceSku{
+				{
+					Name: to.StringPtr("foo"),
+					Kind: to.StringPtr(string(VirtualMachines)),
+					Locations: &[]string{
+						"baz",
+					},
+					LocationInfo: &[]compute.ResourceSkuLocationInfo{
+						{
+							Location: to.StringPtr("baz"),
+							Zones:    &[]string{"1"},
+						},
+					},
+				},
+				{
+					Name: to.StringPtr("foo"),
+					Kind: to.StringPtr(string(VirtualMachines)),
+					Locations: &[]string{
+						"baz",
+					},
+					LocationInfo: &[]compute.ResourceSkuLocationInfo{
+						{
+							Location: to.StringPtr("baz"),
+							Zones:    &[]string{"2"},
+						},
+					},
+				},
+			},
+			want: []string{"1", "2"},
+		},
+		"should not find due to location mismatch": {
+			have: []compute.ResourceSku{
+				{
+					Name: to.StringPtr("foo"),
+					Kind: to.StringPtr(string(VirtualMachines)),
+					Locations: &[]string{
+						"foobar",
+					},
+					LocationInfo: &[]compute.ResourceSkuLocationInfo{
+						{
+							Location: to.StringPtr("foobar"),
+							Zones:    &[]string{"1"},
+						},
+					},
+				},
+			},
+			want: nil,
+		},
+		"should not find due to location restriction": {
+			have: []compute.ResourceSku{
+				{
+					Name: to.StringPtr("foo"),
+					Kind: to.StringPtr(string(VirtualMachines)),
+					Locations: &[]string{
+						"baz",
+					},
+					LocationInfo: &[]compute.ResourceSkuLocationInfo{
+						{
+							Location: to.StringPtr("baz"),
+							Zones:    &[]string{"1"},
+						},
+					},
+					Restrictions: &[]compute.ResourceSkuRestrictions{
+						{
+							Type:   compute.Location,
+							Values: &[]string{"baz"},
+						},
+					},
+				},
+			},
+			want: nil,
+		},
+		"should not find due to zone restriction": {
+			have: []compute.ResourceSku{
+				{
+					Name: to.StringPtr("foo"),
+					Kind: to.StringPtr(string(VirtualMachines)),
+					Locations: &[]string{
+						"baz",
+					},
+					LocationInfo: &[]compute.ResourceSkuLocationInfo{
+						{
+							Location: to.StringPtr("baz"),
+							Zones:    &[]string{"1"},
+						},
+					},
+					Restrictions: &[]compute.ResourceSkuRestrictions{
+						{
+							Type: compute.Zone,
+							RestrictionInfo: &compute.ResourceSkuRestrictionInfo{
+								Zones: &[]string{"1"},
+							},
+						},
+					},
+				},
+			},
+			want: nil,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			// t.Parallel()
```

Comment:
remove?

Reply:
make sure to add `tc := tc` above if you uncomment (see #806, we were missing this in a bunch of places)